Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -0,0 +1,1267 @@
+
+
+def logout(session):
+    """
+    Due to limited amount of simultaneous connections we log out after each API request.
+    Simple post request to /logout endpoint without params.
+    """
+    url_suffix = '/logout'
+    params = {}  # type: dict
+    session.post(SERVER + url_suffix, data=params, verify=USE_SSL)
+
+
+def policy_addr_array_from_arg(policy_addr_data, is_data_string=True):
+    # if the data isn't in string format, it's already an array and requires no formatting
+    policy_adr_str_array = policy_addr_data.split(",") if is_data_string else policy_addr_data
+    policy_addr_dict_array = []
+    for src_addr_name in policy_adr_str_array:
+        cur_addr_dict = {
+            "name": src_addr_name
+        }
+        policy_addr_dict_array.append(cur_addr_dict)
+    return policy_addr_dict_array
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Perform basic login and logout operation, validate connection.
+    """
+    http_request('GET', 'cmdb/system/vdom')
+    return True
+
+
+def get_addresses_command():
+    contents = []
+    context = {}
+    addresses_context = []
+    address = demisto.args().get('address')
+    name = demisto.args().get('name', '')
+
+    addresses = get_addresses_request(address, name)
+    for address in addresses:
+        subnet = address.get('subnet')
+        if subnet:
+            subnet = subnet.replace(" ", "-")
+        contents.append({
+            'Name': address.get('name'),
+            'Subnet': subnet,
+            'StartIP': address.get('start-ip'),
+            'EndIP': address.get('end-ip')
+        })
+        addresses_context.append({
+            'Name': address.get('name'),
+            'Subnet': subnet,
+            'StartIP': address.get('start-ip'),
+            'EndIP': address.get('end-ip')
+        })
+
+    context['Fortigate.Address(val.Name && val.Name === obj.Name)'] = addresses_context
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': contents,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('FortiGate addresses', contents),
+        'EntryContext': context
+    })
+
+
+def get_addresses_request(address, name):
+    uri_suffix = 'cmdb/firewall/address/' + name
+    params = {
+        'vdom': address
+    }
+    response = http_request('GET', uri_suffix, params)
+    # Different structure if we choose all domains
+    if address == '*':
+        return response[0].get('results')
+    return response.get('results')
+
+
+def create_address_command():
+    contents = []
+    context = {}
+    address_context = []
+    address_name = demisto.args().get('name', '')
+    address = demisto.args().get('address', '')
+    mask = demisto.args().get('mask', '')
+    fqdn = demisto.args().get('fqdn', '')
+
+    if fqdn and address:
+        return_error("Please provide only one of the two arguments: fqdn or address")
```

Comment:
What happens if the user does not input value to any of them?
Should it also be an error?

Reply:
Yup will have additional input validation line to check for that